src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,99 @@
+package duke.ui;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+
+import duke.data.exception.DukeException;
+import duke.message.Messages;
+
+
+public class Ui {
+    private static final String DIVIDER = "____________________________________________________________";
+    private final Scanner in;
+    private final PrintStream out;
+
+    public Ui() {
+        this(System.in, System.out);
+    }
+
+    public Ui(InputStream in, PrintStream out) {
+        this.in = new Scanner(in);
+        this.out = out;
```

Comment:
Is there a need for this variable? Seems like its always going to be System.out so why store it in another variable?